pkg/alertmanager/alertmanager.go

```diff
@@ -340,6 +343,43 @@ func (am *Alertmanager) GetReceiversHandler(w http.ResponseWriter, _ *http.Reque
 	}
 }
 
+func (am *Alertmanager) TestReceiversHandler(w http.ResponseWriter, r *http.Request) {
+	payload, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("error reading request body: %s", err.Error()),
+			http.StatusBadRequest)
+	}
+
+	c := alertingNotify.TestReceiversConfigBodyParams{}
+	err = json.Unmarshal(payload, &c)
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("error unmarshalling test receivers config JSON: %s", err.Error()),
+			http.StatusBadRequest)
+	}
+	response, err := alertingNotify.TestReceivers(context.Background(), c, am.cfg.Templates, am.BuildGrafanaReceiverIntegrations, am.cfg.ExternalURL.String())
```

Comment:
We shouldn't be using `context.Background()` here

```suggestion
	response, err := alertingNotify.TestReceivers(r.Context(), c, am.cfg.Templates, am.BuildGrafanaReceiverIntegrations, am.cfg.ExternalURL.String())
```